In [48]:
import sys
import pandas as pd
import numpy as np
from pathlib import Path

sys.modules.pop('a20_config.a10_config', None)  # force reload
from a20_config.a10_config import dyr, file, const, var

Path.cwd()

PosixPath('/sharedata/camm/c_projects/i100_g7__VC_arms')

In [2]:
df_ent_aff_rel = pd.read_parquet(file.pb_ent_aff_rel__parq)
df_ent_aff_rel

,EntityID,AffiliateID,AffiliateName,Industry,Location,YearFounded,AffiliateType,HQCity,HQState_Province,HQCountry,LastUpdated
0,10011-61,52315-12,Ares Management,Asset Manager,"Los Angeles, CA",1997.0,Parent,Los Angeles,California,United States,01/30/2023
1,57113-56,132850-99,Edco Eindhoven,Distributors/Wholesale,"Eindhoven, Netherlands",1978.0,Parent,Eindhoven,None,Netherlands,01/30/2023
2,10010-89,42930-37,Gadzoox Networks,Storage (IT),"San Jose, CA",1992.0,Subsidiary,San Jose,California,United States,01/30/2023
3,57127-24,10358-29,Aurorium,Specialty Chemicals,"Indianapolis, IN",1856.0,Parent,Indianapolis,Indiana,United States,03/14/2023
4,10010-89,123134-23,M-Stream,Media and Information Services (B2B),"Nantes, France",2001.0,Subsidiary,Nantes,None,France,01/30/2023
...,...,...,...,...,...,...,...,...,...,...,...
1622349,482203-99,484327-81,Tandläkarhuset Enköping,Clinics/Outpatient Services,"Enkoping, Sweden",2013.0,Sister,Enkoping,None,Sweden,01/30/2023
1622350,482203-99,455009-77,LudvikaTandläkarna,Clinics/Outpatient Services,"Ludvika, Sweden",2019.0,Sister,Ludvika,None,Sweden,01/30/2023
1622351,482203-99,466763-77,Solna Dental,Clinics/Outpatient Services,"Stockholm, Sweden",2014.0,Sister,Stockholm,None,Sweden,01/30/2023
1622352,482203-99,225754-66,City Dental,Clinics/Outpatient Services,"Stockholm, Sweden",2005.0,Sister,Stockholm,None,Sweden,01/30/2023


In [3]:
df_industry_vals = df_ent_aff_rel[var.industry].value_counts()
df_industry_vals = df_industry_vals.reset_index()
df_industry_vals

,Industry,count
0,Other Services (B2C Non-Financial),120614
1,Other Commercial Services,84958
2,Insurance Brokers,80592
3,Business/Productivity Software,74757
4,Distributors/Wholesale,68031
...,...,...
285,Fundless Sponsor,4
286,Other Limited Partner,3
287,SBIC,2
288,Other Vendor Type,1


In [4]:
_df = df_industry_vals
_df[_df[var.industry].str.contains('venture', case = False, na = False)]

,Industry,count
34,Corporate Venture Capital,9760
71,Venture Capital,4363
234,Not-For-Profit Venture Capital,197


In [5]:
list__vc_industries = {
        0: 'Corporate Venture Capital',
        1: 'Venture Capital',
        2: 'Not-For-Profit Venture Capital',
}

# keep only VC industries as entities
msk = df_ent_aff_rel[var.industry].isin(list__vc_industries.values())
df_ent_aff_rel_vc = df_ent_aff_rel[msk]

df_ent_aff_rel_vc

,EntityID,AffiliateID,AffiliateName,Industry,Location,YearFounded,AffiliateType,HQCity,HQState_Province,HQCountry,LastUpdated
94,10014-31,234561-52,Deloitte Ventures,Corporate Venture Capital,"Toronto, Canada",NaN,Subsidiary,Toronto,Ontario,Canada,03/30/2023
176,10014-49,470966-50,Blackstone Horizon,Venture Capital,"New York, NY",2021.0,Subsidiary,New York,New York,United States,01/30/2023
194,10014-76,107360-38,PricewaterhouseCoopers Venture Partners,Venture Capital,"New York, NY",NaN,Subsidiary,New York,New York,United States,01/30/2023
242,10014-85,123954-58,KPMG Capital,Corporate Venture Capital,"Amstelveen, Netherlands",2013.0,Subsidiary,Amstelveen,None,Netherlands,04/20/2023
282,10015-03,62153-56,Etventure,Corporate Venture Capital,"Berlin, Germany",2010.0,Subsidiary,Berlin,None,Germany,01/30/2023
...,...,...,...,...,...,...,...,...,...,...,...
1621750,482097-97,40998-97,TTC Ventures,Venture Capital,"Cambridge, MA",1996.0,Sister,Cambridge,Massachusetts,United States,05/06/2023
1622144,482138-56,467889-94,CAS Hongta Capital,Corporate Venture Capital,"Nanjing, China",NaN,Sister,Nanjing,Jiangsu,China,02/10/2023
1622206,482167-45,484851-52,Databricks Ventures,Corporate Venture Capital,"San Francisco, CA",2021.0,Sister,San Francisco,California,United States,05/06/2023
1622255,482179-42,40398-58,Chevron Technology Ventures,Corporate Venture Capital,"Houston, TX",1999.0,Sister,Houston,Texas,United States,01/30/2023


In [6]:
df_ent_aff_rel_vc[var.industry].value_counts()

Industry
Corporate Venture Capital         9760
Venture Capital                   4363
Not-For-Profit Venture Capital     197
Name: count, dtype: int64

Until here I have all the VC entities with their affiliated entities.

In [7]:
# Here, I used the PitchBook website for industry categories.
hospital_industries = [
        'Clinics/Outpatient Services',
        'Hospitals/Inpatient Services',
        'Elder and Disabled Care',
        'Laboratory Services (Healthcare)',
        'Other Healthcare Services',
]

Not_considered_hospital_industries : all other industries in group 5 of industry categories except considered ones above (see the website)

? : search for health, hospital, clinic in the industry field:\
     Not needed because I used the PitchBook website for industry categories.

In [8]:
msk = df_ent_aff_rel[var.industry].isin(hospital_industries)
df_hosp = df_ent_aff_rel[msk]
df_hosp

,EntityID,AffiliateID,AffiliateName,Industry,Location,YearFounded,AffiliateType,HQCity,HQState_Province,HQCountry,LastUpdated
13,57177-73,143990-38,Metasoft Systems,Other Healthcare Services,"Vancouver, Canada",1988.0,Parent,Vancouver,British Columbia,Canada,01/30/2023
105,57168-46,56824-39,Vivalto Sante,Clinics/Outpatient Services,"Saint-Grégoire, France",2009.0,Parent,Saint-Grégoire,None,France,01/30/2023
215,57166-75,87044-77,Merieux Nutrisciences,Laboratory Services (Healthcare),"Chicago, IL",1967.0,Parent,Chicago,Illinois,United States,03/08/2023
235,57173-05,52883-92,Aster DM Healthcare,Hospitals/Inpatient Services,"Dubai, United Arab Emirates",1987.0,Parent,Dubai,None,United Arab Emirates,01/30/2023
249,57121-57,439040-44,SecureVideo,Clinics/Outpatient Services,"Alameda, CA",2012.0,Sister,Alameda,California,United States,01/30/2023
...,...,...,...,...,...,...,...,...,...,...,...
1622348,482203-99,455010-22,Wallenborg Group,Clinics/Outpatient Services,"Nacka, Sweden",2015.0,Sister,Nacka,None,Sweden,01/30/2023
1622349,482203-99,484327-81,Tandläkarhuset Enköping,Clinics/Outpatient Services,"Enkoping, Sweden",2013.0,Sister,Enkoping,None,Sweden,01/30/2023
1622350,482203-99,455009-77,LudvikaTandläkarna,Clinics/Outpatient Services,"Ludvika, Sweden",2019.0,Sister,Ludvika,None,Sweden,01/30/2023
1622351,482203-99,466763-77,Solna Dental,Clinics/Outpatient Services,"Stockholm, Sweden",2014.0,Sister,Stockholm,None,Sweden,01/30/2023


In [9]:
df_hosp[var.industry].value_counts()

Industry
Clinics/Outpatient Services         63494
Other Healthcare Services           16074
Elder and Disabled Care              9278
Hospitals/Inpatient Services         6926
Laboratory Services (Healthcare)     6273
Name: count, dtype: int64

In [10]:
# Keep only hospitals with affiliation to VC entities
# Industry value is for the affiliated entity so the hospital ID is the affiliate_id
msk = df_hosp[var.entity_id].isin(df_ent_aff_rel_vc[var.affiliate_id])
df_hosp_with_vc_aff = df_hosp[msk]
# In this df we have hospitals with affiliation to VCs, the VCs IDS are in the EntityID field and the hospital IDs are in the AffiliateID field
df_hosp_with_vc_aff

,EntityID,AffiliateID,AffiliateName,Industry,Location,YearFounded,AffiliateType,HQCity,HQState_Province,HQCountry,LastUpdated
6994,434783-53,166514-86,Ballad Health,Hospitals/Inpatient Services,"Johnson City, TN",NaN,Parent,Johnson City,Tennessee,United States,05/06/2023
12218,57221-29,62802-10,Clinique Saint,Hospitals/Inpatient Services,"Poissy, France",1959.0,Sister,Poissy,None,France,03/02/2023
40286,56040-94,56084-95,MemorialCare Health System,Clinics/Outpatient Services,"Long Beach, CA",1907.0,Parent,Long Beach,California,United States,01/30/2023
41381,433943-74,185286-07,St. John Medical Center (Sisters of Charity He...,Hospitals/Inpatient Services,"Cleveland, OH",NaN,Sister,Cleveland,Ohio,United States,05/06/2023
43802,433710-46,50987-17,Mass General Brigham,Hospitals/Inpatient Services,"Boston, MA",1994.0,Parent,Boston,Massachusetts,United States,05/06/2023
...,...,...,...,...,...,...,...,...,...,...,...
1578190,435625-48,124833-34,East Georgia Regional Medical Center,Clinics/Outpatient Services,"Statesboro, GA",NaN,Sister,Statesboro,Georgia,United States,05/06/2023
1578192,435625-48,99685-00,Renal Services,Clinics/Outpatient Services,"London, United Kingdom",2007.0,Sister,London,England,United Kingdom,05/06/2023
1594867,482380-93,159211-27,Edward-Elmhurst Health,Hospitals/Inpatient Services,"Naperville, IL",2013.0,Parent,Naperville,Illinois,United States,05/06/2023
1596295,435253-51,484448-50,Foundations for Change,Clinics/Outpatient Services,"Peoria, AZ",2017.0,Subsidiary,Peoria,Arizona,United States,01/30/2023


In [11]:
# How many VC affiliated to hospitals
df_hosp_with_vc_aff[var.entity_id].nunique()

105

In [12]:
# How many hospitals have VC affiliations
df_hosp_with_vc_aff[var.affiliate_id].nunique()

218

In [13]:
df_hosp_with_vc_aff[var.aff_type].value_counts()

AffiliateType
Sister        194
Parent         40
Subsidiary      9
Name: count, dtype: int64

Q: If we have 218 hospitals should not we have at least 218 VCs with affiliation to hospitals?\
What are the explanations here?
- Some hospitals have affiliation to the same VC arms, so the number of unique VCs affiliated to hospitals can be less than the number of unique hospitals with VC affiliations
- Sister affiliation relations, when both a hospital and a VC arm is part of a group either both having a common parent or both in the same group?
    - How to find groups? How to distinct groups?
        - Sister companies have always a same parent (I assume this unless I find an example that contradicts this, I mean the parent company exists)
    - How to find example for each of these categories?

https://my.pitchbook.com/profile/57221-29

In [14]:
msk = df_hosp_with_vc_aff[var.aff_type].eq(const.sub)
df1 = df_hosp_with_vc_aff[msk]
df1

,EntityID,AffiliateID,AffiliateName,Industry,Location,YearFounded,AffiliateType,HQCity,HQState_Province,HQCountry,LastUpdated
48911,55879-21,470719-63,Luminox Health,Other Healthcare Services,"Tel Aviv, Israel",2013.0,Subsidiary,Tel Aviv,None,Israel,01/30/2023
60274,55440-55,266429-44,United Concordia Dental,Clinics/Outpatient Services,"Harrisburg, PA",1992.0,Subsidiary,Harrisburg,Pennsylvania,United States,01/30/2023
161284,52247-80,88813-09,Jameda,Other Healthcare Services,"Munich, Germany",2007.0,Subsidiary,Munich,None,Germany,01/30/2023
466801,42282-37,433878-04,PureTech LYT,Other Healthcare Services,"Boston, MA",2018.0,Subsidiary,Boston,Massachusetts,United States,01/30/2023
544521,40675-33,326113-03,Cincinnati Children's Liberty Primary Care,Hospitals/Inpatient Services,"Liberty Township, OH",2017.0,Subsidiary,Liberty Township,Ohio,United States,01/30/2023
905097,170627-05,170674-84,1Trial,Other Healthcare Services,"Jaipur, India",NaN,Subsidiary,Jaipur,Rajasthan,India,01/30/2023
1405876,438429-61,495421-03,Beijing Zekang Hospital,Clinics/Outpatient Services,"Beijing, China",2017.0,Subsidiary,Beijing,None,China,01/30/2023
1596295,435253-51,484448-50,Foundations for Change,Clinics/Outpatient Services,"Peoria, AZ",2017.0,Subsidiary,Peoria,Arizona,United States,01/30/2023
1596299,435253-51,435249-19,Cedar Psychiatry,Clinics/Outpatient Services,"Springville, UT",NaN,Subsidiary,Springville,Utah,United States,01/30/2023


In [15]:
msk = df_hosp_with_vc_aff[var.aff_type].eq(const.par)
df2 = df_hosp_with_vc_aff[msk]
df2

,EntityID,AffiliateID,AffiliateName,Industry,Location,YearFounded,AffiliateType,HQCity,HQState_Province,HQCountry,LastUpdated
6994,434783-53,166514-86,Ballad Health,Hospitals/Inpatient Services,"Johnson City, TN",NaN,Parent,Johnson City,Tennessee,United States,05/06/2023
40286,56040-94,56084-95,MemorialCare Health System,Clinics/Outpatient Services,"Long Beach, CA",1907.0,Parent,Long Beach,California,United States,01/30/2023
43802,433710-46,50987-17,Mass General Brigham,Hospitals/Inpatient Services,"Boston, MA",1994.0,Parent,Boston,Massachusetts,United States,05/06/2023
44669,55757-98,51726-52,Mayo Clinic,Clinics/Outpatient Services,"Rochester, MN",1889.0,Parent,Rochester,Minnesota,United States,01/30/2023
60118,55472-23,10441-90,HCA Management Services,Hospitals/Inpatient Services,"Nashville, TN",1968.0,Parent,Nashville,Tennessee,United States,01/30/2023
65493,149695-84,126530-83,Healthware Group,Other Healthcare Services,"Salerno, Italy",1996.0,Parent,Salerno,None,Italy,05/06/2023
82816,432514-09,55280-53,BGI Genomics,Laboratory Services (Healthcare),"Shenzhen, China",1999.0,Parent,Shenzhen,Guangdong,China,01/30/2023
85463,54810-10,54249-85,UNC REX Healthcare,Laboratory Services (Healthcare),"Raleigh, NC",1894.0,Parent,Raleigh,North Carolina,United States,01/30/2023
185545,51450-31,150350-41,IU Health La Porte Hospital,Hospitals/Inpatient Services,"La Porte, IN",1972.0,Parent,La Porte,Indiana,United States,01/30/2023
242634,521701-84,64677-79,Providence Health Care,Clinics/Outpatient Services,"Vancouver, Canada",1997.0,Parent,Vancouver,British Columbia,Canada,05/06/2023


In [16]:
# VCs that has affiliation to hospitals
msk = df_ent_aff_rel_vc[var.entity_id].isin(df_hosp[var.affiliate_id])
df_vc_with_hosp_aff = df_ent_aff_rel_vc[msk]
df_vc_with_hosp_aff

,EntityID,AffiliateID,AffiliateName,Industry,Location,YearFounded,AffiliateType,HQCity,HQState_Province,HQCountry,LastUpdated
9233,151167-52,399434-32,Healthcare Innovation Fund,Corporate Venture Capital,"Lafayette, LA",NaN,Subsidiary,Lafayette,Louisiana,United States,05/06/2023
28565,433878-04,42282-37,PureTech Health,Venture Capital,"Boston, MA",2001.0,Parent,Boston,Massachusetts,United States,01/30/2023
29682,151100-02,55472-23,Health Insight Capital,Corporate Venture Capital,"Nashville, TN",NaN,Sister,Nashville,Tennessee,United States,01/30/2023
42377,433997-20,168311-71,Sodexo Ventures,Corporate Venture Capital,"Issy les Moulineaux, France",2016.0,Sister,Issy les Moulineaux,None,France,01/30/2023
42882,56084-95,56040-94,MemorialCare Innovation Fund,Corporate Venture Capital,"Long Beach, CA",1993.0,Subsidiary,Long Beach,California,United States,01/30/2023
...,...,...,...,...,...,...,...,...,...,...,...
1608208,10466-20,168311-71,Sodexo Ventures,Corporate Venture Capital,"Issy les Moulineaux, France",2016.0,Sister,Issy les Moulineaux,None,France,01/30/2023
1609901,10326-34,167357-98,Well Ventures,Corporate Venture Capital,"Deerfield, IL",2009.0,Subsidiary,Deerfield,Illinois,United States,01/30/2023
1610837,10308-70,120443-77,McKesson Ventures,Corporate Venture Capital,"San Francisco, CA",2014.0,Sister,San Francisco,California,United States,01/30/2023
1615178,152034-13,166204-27,Fresenius Medical Care Ventures,Corporate Venture Capital,"Bad Homburg, Germany",2016.0,Sister,Bad Homburg,None,Germany,01/30/2023


In [17]:
# how many hospitals affiliated to VCs
df_vc_with_hosp_aff[var.entity_id].nunique()

218

In [18]:
# how many VCs have hospital affiliations
df_vc_with_hosp_aff[var.affiliate_id].nunique()

109

In [19]:
df_vc_with_hosp_aff[var.aff_type].value_counts()

AffiliateType
Sister        195
Subsidiary     43
Parent          8
Name: count, dtype: int64

The question is why are not the same number? isn't the affiliaton relation bidirectional? I mean if A is affiliated to B then B is affiliated to A?

What I have done for the first df_hosp_with_vc_aff?

In [20]:
# python
# detect asymmetric pairs (A->B present but B->A missing)
pairs = set(zip(df_ent_aff_rel[var.entity_id], df_ent_aff_rel[var.affiliate_id]))
rev_pairs = set((b, a) for a, b in pairs)

asymmetric = pairs - rev_pairs

len(pairs), len(asymmetric), list(asymmetric)[:5]

(1622226,
 41949,
 [('156821-86', '112202-20'),
  ('42716-44', '167987-89'),
  ('63543-52', '64465-39'),
  ('160890-22', '58968-82'),
  ('407210-32', '52611-85')])

In [21]:
df_hosp_with_vc_aff.to_excel(file.di_sp050_05__hospitals_with_affiliated_vc_by_data_properties_only__xl, index = False)
df_vc_with_hosp_aff.to_excel(file.di_sp050_10__VCs_with_affiliated_hospitals_by_data_properties_only__xl, index = False)

# ADDING NOT INCLUDED PREVIOUSLY MANUALLY FOUND HOSPITALS AND VCs
Now I have to make sure that all the previously found hospitals and VCs are included in this

In [23]:
# Are all the previous VC And hospitals found available in this data?
df_prev_hosp_aff_ent = pd.read_excel(file.dm_sp040_10__affiliated_entities_manual_list__xl)
df_prev_hosp_aff_ent

,EntityID,AffiliateID,AffiliateName,Industry,Location,YearFounded,AffiliateType,HQCity,HQState_Province,HQCountry,LastUpdated
0,151167-52,399434-32,Healthcare Innovation Fund,Corporate Venture Capital,"Lafayette, LA",NaN,Subsidiary,Lafayette,Louisiana,United States,05/06/2023
1,56084-95,56040-94,MemorialCare Innovation Fund,Corporate Venture Capital,"Long Beach, CA",1993.0,Subsidiary,Long Beach,California,United States,01/30/2023
2,56084-95,62952-67,Summation Health Ventures,Corporate Venture Capital,"Long Beach, CA",2014.0,Subsidiary,Long Beach,California,United States,01/30/2023
3,150350-41,51450-31,CHV Capital,Venture Capital,"Indianapolis, IN",2007.0,Subsidiary,Indianapolis,Indiana,United States,01/30/2023
4,55165-33,433943-74,University Hospitals Ventures,Corporate Venture Capital,"Cleveland, OH",2018.0,Subsidiary,Cleveland,Ohio,United States,03/03/2023
5,51614-92,11114-65,Ascension Ventures,Corporate Venture Capital,"Saint Louis, MO",2001.0,Subsidiary,Saint Louis,Missouri,United States,01/30/2023
6,51726-52,55757-98,Mayo Clinic Ventures,Venture Capital,"Rochester, MN",1986.0,Subsidiary,Rochester,Minnesota,United States,01/30/2023
7,50987-17,118715-50,Mass General Brigham Innovation,Corporate Venture Capital,"Cambridge, MA",NaN,Subsidiary,Cambridge,Massachusetts,United States,01/30/2023
8,50987-17,433710-46,Mass General Brigham Ventures,Venture Capital,"Cambridge, MA",2008.0,Subsidiary,Cambridge,Massachusetts,United States,04/24/2023
9,185503-60,266690-26,Intermountain Ventures,Corporate Venture Capital,"Salt Lake City, UT",2012.0,Subsidiary,Salt Lake City,Utah,United States,01/30/2023


In [24]:
all_pre_ids = pd.concat([
        df_prev_hosp_aff_ent[var.entity_id],
        df_prev_hosp_aff_ent[var.affiliate_id],
]).unique()
all_pre_ids.size, all_pre_ids[:5]

(38,
 array(['151167-52', '56084-95', '150350-41', '55165-33', '51614-92'],
       dtype=object))

In [25]:
all_new_ids = pd.concat([
        df_hosp_with_vc_aff[var.entity_id],
        df_hosp_with_vc_aff[var.affiliate_id],
        df_vc_with_hosp_aff[var.entity_id],
        df_vc_with_hosp_aff[var.affiliate_id],
]).unique()
all_new_ids.size, all_new_ids[:5]

(331,
 array(['434783-53', '57221-29', '56040-94', '433943-74', '433710-46'],
       dtype=object))

In [26]:
# check if all previous ids are in the new ids
set(all_pre_ids).issubset(set(all_new_ids))

False

In [27]:
# find the missing ids
missing_ids = set(all_pre_ids) - set(all_new_ids)
len(missing_ids), list(missing_ids)[:5]

(8, ['221543-56', '268046-92', '340845-76', '64197-64', '61215-67'])

Add these missing ids to the VC ids list or hospital ids list based on their type
first I have to find that each of these ids is VC or hospital, those ids that are in the hospital.xlsx created before are hospitals, the rest are VCs

In [28]:
df_hosp_c030 = pd.read_excel(file.dm_sp040_15__manual_hospitals_list__xl)
df_hosp_c030

,NameFromMedia,PBID,PBURL,PBEntityType,Company.CompanyID,Company.CompanyName,Company.CompanyAlsoKnownAs,Company.CompanyFormerName,Company.CompanyLegalName,Company.Description,...,LimitedPartner.TargetFixedIncomeMin,LimitedPartner.TargetFixedIncomePercentMin,LimitedPartner.TargetFixedIncomeMax,LimitedPartner.TargetFixedIncomePercentMax,LimitedPartner.TargetCashMin,LimitedPartner.TargetCashPercentMin,LimitedPartner.TargetCashMax,LimitedPartner.TargetCashPercentMax,LimitedPartner.LastUpdated,Matched_From_LP
0,Ascension (St. Louis): Ascension Ventures,51614-92,https://my.pitchbook.com/profile/51614-92/comp...,com,51614-92,Ascension Health,Ascension,NaN,NaN,Provider of a health system intended to provid...,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,01/30/2023,1.0
1,"Ballad Health (Johnson City, Tenn.): Ballad Ve...",166514-86,https://my.pitchbook.com/profile/166514-86/com...,com,166514-86,Ballad Health,Better Together,"Wellmont Health System, Mountain States Health...",NaN,Provider of healthcare services intended to of...,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,Cleveland Clinic: Cleveland Clinic Ventures,127260-28,https://my.pitchbook.com/profile/127260-28/inv...,inv,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,"Elmhurst Memorial Hospital (Warrenville, Ill.)...",159211-27,https://my.pitchbook.com/profile/159211-27/com...,com,159211-27,Edward-Elmhurst Health,"EEH, Edward-Elmhurst",NaN,NaN,Operator of general medical and surgical hospi...,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4,Henry Ford Health System (Detroit): Henry Ford...,100427-41,https://my.pitchbook.com/profile/100427-41/com...,com,100427-41,Henry Ford Health System,"Henry Ford, HFHS",NaN,"Henry Ford Health System, Inc.",Henry Ford Health System is a health care prov...,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
5,Intermountain Healthcare (Salt Lake City): Int...,185503-60,https://my.pitchbook.com/profile/185503-60/com...,com,185503-60,Intermountain Healthcare,Intermountain,NaN,"Intermountain Healthcare, Inc.",Operator of a non-profit healthcare system of ...,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
6,Indiana University Health (Indianapolis): CHV ...,150350-41,https://my.pitchbook.com/profile/150350-41/com...,com,150350-41,IU Health La Porte Hospital,NaN,NaN,NaN,Operator of a hospital. The company owns and o...,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
7,Jefferson Health (Philadelphia): Jefferson Inn...,188107-03,https://my.pitchbook.com/profile/188107-03/inv...,inv,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
8,"Kaiser Permanente (Oakland, Calif.): Kaiser Pe...",229125-07,https://my.pitchbook.com/profile/229125-07/inv...,inv,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
9,Ochsner Lafayette (La.) General Health: Health...,151167-52,https://my.pitchbook.com/profile/151167-52/inv...,inv,151167-52,Ochsner Lafayette General,NaN,Lafayette General Health,"Lafayette General Health System, Inc.",Provider of not-for-profit healthcare services...,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [29]:
# check which of the missing ids are in the hospital data
missing_hosp_ids = set(df_hosp_c030[var.pb_id]).intersection(missing_ids)
len(missing_hosp_ids), missing_hosp_ids

(5, {'127260-28', '55165-33', '58870-45', '61215-67', '64197-64'})

In [30]:
missing_vc_ids = missing_ids - missing_hosp_ids
len(missing_vc_ids), missing_vc_ids

(3, {'221543-56', '268046-92', '340845-76'})

What is the industry of the missing ones? why they are missing?

In [31]:
msk = df_ent_aff_rel[var.affiliate_id].isin(missing_hosp_ids)
df_missd_h = df_ent_aff_rel[msk]

df_missd_h

,EntityID,AffiliateID,AffiliateName,Industry,Location,YearFounded,AffiliateType,HQCity,HQState_Province,HQCountry,LastUpdated
28857,433943-74,55165-33,University Hospitals,Acquirer,None,1866.0,Parent,None,None,None,05/06/2023
61542,244258-57,58870-45,University of Pittsburgh Medical Center,Investor,"Pittsburgh, PA",1986.0,Parent,Pittsburgh,Pennsylvania,United States,05/06/2023
102950,290328-13,58870-45,University of Pittsburgh Medical Center,Investor,"Pittsburgh, PA",1986.0,Parent,Pittsburgh,Pennsylvania,United States,03/16/2023
305081,188063-65,58870-45,University of Pittsburgh Medical Center,Investor,"Pittsburgh, PA",1986.0,Parent,Pittsburgh,Pennsylvania,United States,05/06/2023
360132,92342-62,58870-45,University of Pittsburgh Medical Center,Investor,"Pittsburgh, PA",1986.0,Parent,Pittsburgh,Pennsylvania,United States,01/30/2023
421516,185286-07,55165-33,University Hospitals,Acquirer,None,1866.0,Parent,None,None,None,01/30/2023
787365,221543-56,64197-64,NewYork-Presbyterian Hospital,Foundation,"New York, NY",1998.0,Parent,New York,New York,United States,01/30/2023
806565,268046-92,61215-67,UnityPoint Health,Corporation,"West Des Moines, IA",1994.0,Parent,West Des Moines,Iowa,United States,05/06/2023
982838,357417-64,127260-28,Cleveland Clinic,Acquirer,"Cleveland, OH",1921.0,Parent,Cleveland,Ohio,United States,01/30/2023
1143278,209544-22,58870-45,University of Pittsburgh Medical Center,Investor,"Pittsburgh, PA",1986.0,Parent,Pittsburgh,Pennsylvania,United States,05/06/2023


In [32]:
df_missd_h.drop_duplicates(subset = [var.affiliate_id])

,EntityID,AffiliateID,AffiliateName,Industry,Location,YearFounded,AffiliateType,HQCity,HQState_Province,HQCountry,LastUpdated
28857,433943-74,55165-33,University Hospitals,Acquirer,None,1866.0,Parent,None,None,None,05/06/2023
61542,244258-57,58870-45,University of Pittsburgh Medical Center,Investor,"Pittsburgh, PA",1986.0,Parent,Pittsburgh,Pennsylvania,United States,05/06/2023
787365,221543-56,64197-64,NewYork-Presbyterian Hospital,Foundation,"New York, NY",1998.0,Parent,New York,New York,United States,01/30/2023
806565,268046-92,61215-67,UnityPoint Health,Corporation,"West Des Moines, IA",1994.0,Parent,West Des Moines,Iowa,United States,05/06/2023
982838,357417-64,127260-28,Cleveland Clinic,Acquirer,"Cleveland, OH",1921.0,Parent,Cleveland,Ohio,United States,01/30/2023


In [33]:
msk = df_ent_aff_rel[var.affiliate_id].isin(missing_vc_ids)
df_missd_vc = df_ent_aff_rel[msk]
df_missd_vc

,EntityID,AffiliateID,AffiliateName,Industry,Location,YearFounded,AffiliateType,HQCity,HQState_Province,HQCountry,LastUpdated
785095,127260-28,340845-76,Cleveland Clinic Ventures,Corporate Venture Capital,"Cleveland, OH",2016.0,Subsidiary,Cleveland,Ohio,United States,01/30/2023
997732,357417-64,340845-76,Cleveland Clinic Ventures,Corporate Venture Capital,"Cleveland, OH",2016.0,Sister,Cleveland,Ohio,United States,01/30/2023
1290557,64197-64,221543-56,NewYork-Presbyterian Ventures,Corporate Venture Capital,"New York, NY",2015.0,Subsidiary,New York,New York,United States,01/30/2023
1388227,61215-67,268046-92,UnityPoint Health Ventures,Corporate Venture Capital,"West Des Moines, IA",2019.0,Subsidiary,West Des Moines,Iowa,United States,01/30/2023


In [34]:
df_missd_vc.drop_duplicates(subset = [var.affiliate_id])

,EntityID,AffiliateID,AffiliateName,Industry,Location,YearFounded,AffiliateType,HQCity,HQState_Province,HQCountry,LastUpdated
785095,127260-28,340845-76,Cleveland Clinic Ventures,Corporate Venture Capital,"Cleveland, OH",2016.0,Subsidiary,Cleveland,Ohio,United States,01/30/2023
1290557,64197-64,221543-56,NewYork-Presbyterian Ventures,Corporate Venture Capital,"New York, NY",2015.0,Subsidiary,New York,New York,United States,01/30/2023
1388227,61215-67,268046-92,UnityPoint Health Ventures,Corporate Venture Capital,"West Des Moines, IA",2019.0,Subsidiary,West Des Moines,Iowa,United States,01/30/2023


In [35]:
'Corporate Venture Capital' == list__vc_industries[0]

True

Because the industry of these missing VCs is 'Corporate Venture Capital' they must be included in the VCs data frame:
testing for that:

In [36]:
df_missd_vc[var.industry].isin(df_ent_aff_rel_vc[var.industry]).all()

np.True_

Yes, they are all in the VC industries list as expected
so why they are missing? They might be missing because their affiliated entities are not hospitals based on the industry categories
like the 5 hospitals that are missing or they are missing because their affiliation relations are not in the data (the data is not complete)

In [37]:
hosp_ids_combined = pd.concat([
        df_hosp[var.affiliate_id],
        pd.Series(list(missing_hosp_ids), name = var.affiliate_id),
]).unique()
len(hosp_ids_combined), hosp_ids_combined

(8964,
 array(['143990-38', '56824-39', '87044-77', ..., '127260-28', '58870-45',
        '55165-33'], shape=(8964,), dtype=object))

In [38]:
vc_ids_combined = pd.concat([
        df_ent_aff_rel_vc[var.affiliate_id],
        pd.Series(list(missing_vc_ids), name = var.affiliate_id),
]).unique()
len(vc_ids_combined), vc_ids_combined

(3008,
 array(['234561-52', '470966-50', '107360-38', ..., '482393-44',
        '470596-15', '507015-46'], shape=(3008,), dtype=object))

In [39]:
msk = df_ent_aff_rel[var.affiliate_id].isin(hosp_ids_combined)
df_hosp_2 = df_ent_aff_rel[msk]
df_hosp_2

,EntityID,AffiliateID,AffiliateName,Industry,Location,YearFounded,AffiliateType,HQCity,HQState_Province,HQCountry,LastUpdated
13,57177-73,143990-38,Metasoft Systems,Other Healthcare Services,"Vancouver, Canada",1988.0,Parent,Vancouver,British Columbia,Canada,01/30/2023
105,57168-46,56824-39,Vivalto Sante,Clinics/Outpatient Services,"Saint-Grégoire, France",2009.0,Parent,Saint-Grégoire,None,France,01/30/2023
215,57166-75,87044-77,Merieux Nutrisciences,Laboratory Services (Healthcare),"Chicago, IL",1967.0,Parent,Chicago,Illinois,United States,03/08/2023
235,57173-05,52883-92,Aster DM Healthcare,Hospitals/Inpatient Services,"Dubai, United Arab Emirates",1987.0,Parent,Dubai,None,United Arab Emirates,01/30/2023
249,57121-57,439040-44,SecureVideo,Clinics/Outpatient Services,"Alameda, CA",2012.0,Sister,Alameda,California,United States,01/30/2023
...,...,...,...,...,...,...,...,...,...,...,...
1622348,482203-99,455010-22,Wallenborg Group,Clinics/Outpatient Services,"Nacka, Sweden",2015.0,Sister,Nacka,None,Sweden,01/30/2023
1622349,482203-99,484327-81,Tandläkarhuset Enköping,Clinics/Outpatient Services,"Enkoping, Sweden",2013.0,Sister,Enkoping,None,Sweden,01/30/2023
1622350,482203-99,455009-77,LudvikaTandläkarna,Clinics/Outpatient Services,"Ludvika, Sweden",2019.0,Sister,Ludvika,None,Sweden,01/30/2023
1622351,482203-99,466763-77,Solna Dental,Clinics/Outpatient Services,"Stockholm, Sweden",2014.0,Sister,Stockholm,None,Sweden,01/30/2023


In [40]:
# keep only those hospitals that are affiliated to VCs
msk = df_hosp_2[var.entity_id].isin(vc_ids_combined)
df_hosp_with_vc_aff_2 = df_hosp_2[msk]
df_hosp_with_vc_aff_2

,EntityID,AffiliateID,AffiliateName,Industry,Location,YearFounded,AffiliateType,HQCity,HQState_Province,HQCountry,LastUpdated
6994,434783-53,166514-86,Ballad Health,Hospitals/Inpatient Services,"Johnson City, TN",NaN,Parent,Johnson City,Tennessee,United States,05/06/2023
12218,57221-29,62802-10,Clinique Saint,Hospitals/Inpatient Services,"Poissy, France",1959.0,Sister,Poissy,None,France,03/02/2023
28857,433943-74,55165-33,University Hospitals,Acquirer,None,1866.0,Parent,None,None,None,05/06/2023
40286,56040-94,56084-95,MemorialCare Health System,Clinics/Outpatient Services,"Long Beach, CA",1907.0,Parent,Long Beach,California,United States,01/30/2023
41381,433943-74,185286-07,St. John Medical Center (Sisters of Charity He...,Hospitals/Inpatient Services,"Cleveland, OH",NaN,Sister,Cleveland,Ohio,United States,05/06/2023
...,...,...,...,...,...,...,...,...,...,...,...
1578192,435625-48,99685-00,Renal Services,Clinics/Outpatient Services,"London, United Kingdom",2007.0,Sister,London,England,United Kingdom,05/06/2023
1591445,340845-76,127260-28,Cleveland Clinic,Acquirer,"Cleveland, OH",1921.0,Parent,Cleveland,Ohio,United States,01/30/2023
1594867,482380-93,159211-27,Edward-Elmhurst Health,Hospitals/Inpatient Services,"Naperville, IL",2013.0,Parent,Naperville,Illinois,United States,05/06/2023
1596295,435253-51,484448-50,Foundations for Change,Clinics/Outpatient Services,"Peoria, AZ",2017.0,Subsidiary,Peoria,Arizona,United States,01/30/2023


In [41]:
msk = df_ent_aff_rel[var.affiliate_id].isin(vc_ids_combined)
df_vc_2 = df_ent_aff_rel[msk]
df_vc_2

,EntityID,AffiliateID,AffiliateName,Industry,Location,YearFounded,AffiliateType,HQCity,HQState_Province,HQCountry,LastUpdated
94,10014-31,234561-52,Deloitte Ventures,Corporate Venture Capital,"Toronto, Canada",NaN,Subsidiary,Toronto,Ontario,Canada,03/30/2023
176,10014-49,470966-50,Blackstone Horizon,Venture Capital,"New York, NY",2021.0,Subsidiary,New York,New York,United States,01/30/2023
194,10014-76,107360-38,PricewaterhouseCoopers Venture Partners,Venture Capital,"New York, NY",NaN,Subsidiary,New York,New York,United States,01/30/2023
242,10014-85,123954-58,KPMG Capital,Corporate Venture Capital,"Amstelveen, Netherlands",2013.0,Subsidiary,Amstelveen,None,Netherlands,04/20/2023
282,10015-03,62153-56,Etventure,Corporate Venture Capital,"Berlin, Germany",2010.0,Subsidiary,Berlin,None,Germany,01/30/2023
...,...,...,...,...,...,...,...,...,...,...,...
1621750,482097-97,40998-97,TTC Ventures,Venture Capital,"Cambridge, MA",1996.0,Sister,Cambridge,Massachusetts,United States,05/06/2023
1622144,482138-56,467889-94,CAS Hongta Capital,Corporate Venture Capital,"Nanjing, China",NaN,Sister,Nanjing,Jiangsu,China,02/10/2023
1622206,482167-45,484851-52,Databricks Ventures,Corporate Venture Capital,"San Francisco, CA",2021.0,Sister,San Francisco,California,United States,05/06/2023
1622255,482179-42,40398-58,Chevron Technology Ventures,Corporate Venture Capital,"Houston, TX",1999.0,Sister,Houston,Texas,United States,01/30/2023


In [42]:
# keep only those affiliation of these VCs that the affiliated entity (i.e. Entitiy ID not the AffiliateID) is a Hospital
msk = df_vc_2[var.entity_id].isin(hosp_ids_combined)
df_vc_with_hosp_aff_2 = df_vc_2[msk]
df_vc_with_hosp_aff_2

,EntityID,AffiliateID,AffiliateName,Industry,Location,YearFounded,AffiliateType,HQCity,HQState_Province,HQCountry,LastUpdated
9233,151167-52,399434-32,Healthcare Innovation Fund,Corporate Venture Capital,"Lafayette, LA",NaN,Subsidiary,Lafayette,Louisiana,United States,05/06/2023
28565,433878-04,42282-37,PureTech Health,Venture Capital,"Boston, MA",2001.0,Parent,Boston,Massachusetts,United States,01/30/2023
29682,151100-02,55472-23,Health Insight Capital,Corporate Venture Capital,"Nashville, TN",NaN,Sister,Nashville,Tennessee,United States,01/30/2023
42377,433997-20,168311-71,Sodexo Ventures,Corporate Venture Capital,"Issy les Moulineaux, France",2016.0,Sister,Issy les Moulineaux,None,France,01/30/2023
42882,56084-95,56040-94,MemorialCare Innovation Fund,Corporate Venture Capital,"Long Beach, CA",1993.0,Subsidiary,Long Beach,California,United States,01/30/2023
...,...,...,...,...,...,...,...,...,...,...,...
1608208,10466-20,168311-71,Sodexo Ventures,Corporate Venture Capital,"Issy les Moulineaux, France",2016.0,Sister,Issy les Moulineaux,None,France,01/30/2023
1609901,10326-34,167357-98,Well Ventures,Corporate Venture Capital,"Deerfield, IL",2009.0,Subsidiary,Deerfield,Illinois,United States,01/30/2023
1610837,10308-70,120443-77,McKesson Ventures,Corporate Venture Capital,"San Francisco, CA",2014.0,Sister,San Francisco,California,United States,01/30/2023
1615178,152034-13,166204-27,Fresenius Medical Care Ventures,Corporate Venture Capital,"Bad Homburg, Germany",2016.0,Sister,Bad Homburg,None,Germany,01/30/2023


In [43]:
# how many hospitals with VC affiliations now
df_hosp_with_vc_aff_2[var.affiliate_id].nunique()

223

In [44]:
# how many VCs with hospital affiliations now
df_hosp_with_vc_aff_2[var.entity_id].nunique()

108

In [45]:
# how many VCs with hospital affiliations now
df_vc_with_hosp_aff_2[var.affiliate_id].nunique()

112

In [46]:
df_vc_with_hosp_aff_2[var.entity_id].nunique()

223

In [49]:
# save new data
df_hosp_with_vc_aff_2.to_excel(
        file.di_sp050_15__hospitals_with_affiliated_vc__manual_list_added_to_by_data_property__xl,
        index = False)

df_vc_with_hosp_aff_2.to_excel(
        file.di_sp050_20__VCs_with_affiliated_hospitals_manual_list_added_to_by_data_property__xl,
        index = False)

---

# GETTING SOME STATS

In [50]:
# just for checking the missing ids are now included
all_new_ids_2 = pd.concat([
        df_hosp_with_vc_aff_2[var.entity_id],
        df_hosp_with_vc_aff_2[var.affiliate_id],
        df_vc_with_hosp_aff_2[var.entity_id],
        df_vc_with_hosp_aff_2[var.affiliate_id],
]).unique()

set(all_pre_ids).issubset(set(all_new_ids_2))


True

In [51]:
# now they are all included

In [52]:
unique_vc_ids_final = pd.concat([
        df_hosp_with_vc_aff_2[var.entity_id],
        df_vc_with_hosp_aff_2[var.affiliate_id],
]).unique()
unique_vc_ids_final.size

113

In [53]:
unique_hosp_ids_final = pd.concat([
        df_hosp_with_vc_aff_2[var.affiliate_id],
        df_vc_with_hosp_aff_2[var.entity_id],
]).unique()
unique_hosp_ids_final.size

226

In [54]:
df_hosp_with_vc_aff_2[var.aff_type].value_counts()

AffiliateType
Sister        194
Parent         45
Subsidiary      9
Name: count, dtype: int64

In [55]:
df_vc_with_hosp_aff_2[var.aff_type].value_counts()

AffiliateType
Sister        195
Subsidiary     48
Parent          8
Name: count, dtype: int64

What is the exact number of VCs that are subsidiay to a hospital?
- The related question is what is the direction of affiliation between hospitals and VCs?

- What is the direction of aff_type?

    The affilate id has the stated affilate type to the entity, so if the affilate type is subsidiary then if the affiliate id is a hospital then it is a subsidiary to the entity id which is a VC entity.

In [56]:
msk = df_hosp_with_vc_aff_2[var.aff_type].eq(const.sub)

df_a = df_hosp_with_vc_aff_2[msk]
df_a

,EntityID,AffiliateID,AffiliateName,Industry,Location,YearFounded,AffiliateType,HQCity,HQState_Province,HQCountry,LastUpdated
48911,55879-21,470719-63,Luminox Health,Other Healthcare Services,"Tel Aviv, Israel",2013.0,Subsidiary,Tel Aviv,None,Israel,01/30/2023
60274,55440-55,266429-44,United Concordia Dental,Clinics/Outpatient Services,"Harrisburg, PA",1992.0,Subsidiary,Harrisburg,Pennsylvania,United States,01/30/2023
161284,52247-80,88813-09,Jameda,Other Healthcare Services,"Munich, Germany",2007.0,Subsidiary,Munich,None,Germany,01/30/2023
466801,42282-37,433878-04,PureTech LYT,Other Healthcare Services,"Boston, MA",2018.0,Subsidiary,Boston,Massachusetts,United States,01/30/2023
544521,40675-33,326113-03,Cincinnati Children's Liberty Primary Care,Hospitals/Inpatient Services,"Liberty Township, OH",2017.0,Subsidiary,Liberty Township,Ohio,United States,01/30/2023
905097,170627-05,170674-84,1Trial,Other Healthcare Services,"Jaipur, India",NaN,Subsidiary,Jaipur,Rajasthan,India,01/30/2023
1405876,438429-61,495421-03,Beijing Zekang Hospital,Clinics/Outpatient Services,"Beijing, China",2017.0,Subsidiary,Beijing,None,China,01/30/2023
1596295,435253-51,484448-50,Foundations for Change,Clinics/Outpatient Services,"Peoria, AZ",2017.0,Subsidiary,Peoria,Arizona,United States,01/30/2023
1596299,435253-51,435249-19,Cedar Psychiatry,Clinics/Outpatient Services,"Springville, UT",NaN,Subsidiary,Springville,Utah,United States,01/30/2023


In [58]:
msk = df_vc_with_hosp_aff_2[var.aff_type].eq(const.par)

df_b = df_vc_with_hosp_aff_2[msk]
df_b

,EntityID,AffiliateID,AffiliateName,Industry,Location,YearFounded,AffiliateType,HQCity,HQState_Province,HQCountry,LastUpdated
28565,433878-04,42282-37,PureTech Health,Venture Capital,"Boston, MA",2001.0,Parent,Boston,Massachusetts,United States,01/30/2023
327833,470719-63,55879-21,OurCrowd,Venture Capital,"Jerusalem, Israel",2013.0,Parent,Jerusalem,None,Israel,01/30/2023
446639,326113-03,40675-33,Cincinnati Children's Hospital Medical Center,Not-For-Profit Venture Capital,"Cincinnati, OH",1883.0,Parent,Cincinnati,Ohio,United States,05/06/2023
500438,88813-09,52247-80,Burda Digital Ventures,Corporate Venture Capital,"Munich, Germany",1999.0,Parent,Munich,None,Germany,01/30/2023
905161,170674-84,170627-05,CaseAtlas,Venture Capital,"New York, NY",2012.0,Parent,New York,New York,United States,01/30/2023
1083372,495421-03,438429-61,Beijing Financial Street Capital Operation Center,Venture Capital,"Beijing, China",2012.0,Parent,Beijing,None,China,05/11/2023
1471035,484448-50,435253-51,Novamind,Venture Capital,"Toronto, Canada",2019.0,Parent,Toronto,Ontario,Canada,01/30/2023
1593627,435249-19,435253-51,Novamind,Venture Capital,"Toronto, Canada",2019.0,Parent,Toronto,Ontario,Canada,01/30/2023
